tests/core/daemon/test_daemon.py

```diff
@@ -823,3 +875,279 @@ async def test_unexpected_json(
+    response_data_dict: Dict[str, Any],
+) -> None:
+    ws, keychain = daemon_connection_and_temp_keychain
+
+    keychain.set_master_passphrase(
+        current_passphrase=DEFAULT_PASSPHRASE_IF_NO_MASTER_PASSPHRASE, new_passphrase="this is a passphrase"
+    )
+
+    payload = create_payload(
+        method,
+        parameter,
+        "service_name",
+        "daemon",
+    )
+    await ws.send_str(payload)
+    response = await ws.receive()
+
+    assert_response(response, response_data_dict)
+
+
+@pytest.mark.parametrize(
+    "method, parameter, response_data_dict",
+    [
+        (
+            "unlock_keyring",
+            {"key": "wrong passphrase"},
+            {"success": False, "error": "bad passphrase"},
+        ),
+        (
+            "unlock_keyring",
+            {"key": True},
+            {"success": False, "error": "missing key"},
+        ),
+        (
+            "unlock_keyring",
+            {},
+            {"success": False, "error": "missing key"},
+        ),
+        (
+            "unlock_keyring",
+            {"key": "this is a passphrase"},
+            {"success": True, "error": None},
+        ),
+    ],
+    ids=["wrong current passphrase", "incorrect type", "missing data", "correct"],
+)
+@pytest.mark.asyncio
+async def test_unlock_keyring(
+    daemon_connection_and_temp_keychain: Tuple[aiohttp.ClientWebSocketResponse, Keychain],
+    method: str,
+    parameter: Dict[str, Any],
+    response_data_dict: Dict[str, Any],
+) -> None:
+    ws, keychain = daemon_connection_and_temp_keychain
+
+    keychain.set_master_passphrase(
+        current_passphrase=DEFAULT_PASSPHRASE_IF_NO_MASTER_PASSPHRASE, new_passphrase="this is a passphrase"
+    )
+
+    payload = create_payload(
+        method,
+        parameter,
+        "service_name",
+        "daemon",
+    )
+    await ws.send_str(payload)
+    response = await ws.receive()
+
+    assert_response(response, response_data_dict)
+
+
+@pytest.mark.parametrize(
+    "method, parameter, response_data_dict",
+    [
+        (
+            "unlock_keyring",
+            {"key": "this is a passphrase"},
+            {"success": False, "error": "validation exception"},
+        ),
+        (
+            "validate_keyring_passphrase",
+            {"key": "this is a passphrase"},
+            {"success": False, "error": "validation exception"},
+        ),
+    ],
+    ids=["unlock_keyring", "validate_keyring_passphrase"],
+)
+@pytest.mark.asyncio
+async def test_keyring_file_deleted(
+    daemon_connection_and_temp_keychain: Tuple[aiohttp.ClientWebSocketResponse, Keychain],
+    method: str,
+    parameter: Dict[str, Any],
+    response_data_dict: Dict[str, Any],
+) -> None:
+    ws, keychain = daemon_connection_and_temp_keychain
+
+    keychain.set_master_passphrase(
+        current_passphrase=DEFAULT_PASSPHRASE_IF_NO_MASTER_PASSPHRASE, new_passphrase="this is a passphrase"
+    )
+    os.remove(keychain.keyring_wrapper.keyring.keyring_path)
```

Comment:
https://docs.python.org/3.11/library/pathlib.html#pathlib.Path.unlink

```suggestion
    keychain.keyring_wrapper.keyring.keyring_path.unlink()
```

Reply:
done